modules/tinymce/src/themes/silver/test/ts/browser/editor/bespoke/NamedStyleSelectFormatsTest.ts

```diff
@@ -0,0 +1,101 @@
+import Theme from 'tinymce/themes/silver/Theme';
+import * as MenuUtils from '../../../module/MenuUtils';
+
+interface StyleSelectMenuItem {
+  element: string;
+  title: string;
+}
+
+UnitTest.asynctest('browser.tinymce.themes.silver.bespoke.NamedStyleSelectFormatTest', (success, failure) => {
+  Theme();
+  TinyLoader.setup(
+    (editor, onSuccess, onFailure) => {
+      const sAssertStyleSelectMenuItems = (label: string, expectedItems: StyleSelectMenuItem[]) => Logger.t(
+        label,
+        Chain.asStep(SugarBody.body(), [
+          UiFinder.cFindIn('.tox-selected-menu .tox-collection__group'),
+          Assertions.cAssertStructure('Checking structure', ApproxStructure.build((s, str, arr) => s.element('div', {
+            classes: [ arr.has('tox-collection__group') ],
+            children: Arr.map(expectedItems, (expected) => s.element('div', {
+              attrs: {
+                role: str.is('menuitemcheckbox'),
+                title: str.is(expected.title)
+              },
+              children: [
+                s.element('div', {
+                  classes: [ arr.has('tox-collection__item-label') ],
+                  children: [
+                    s.element(expected.element, {
+                      children: [
+                        s.text(str.is(expected.title))
+                      ]
+                    })
+                  ]
+                }),
+                s.element('div', {
+                  classes: [ arr.has('tox-collection__item-checkmark') ],
+                  children: [
+                    s.anything()
+                  ]
+                })
+              ]
+            }))
+          })))
+        ])
+      );
+
+      const sAssertFormatExists = (formatName: string) =>
+        Step.sync(() =>
+          Assert.eq(`Expected format: ${formatName} to exist`, true, editor.formatter.has(formatName))
+        );
+
+      Pipeline.async({ }, [
+        MenuUtils.sOpenMenu('Format', 'Paragraph:last'),
+        sAssertStyleSelectMenuItems('Checking style select items', [
+          { title: 'My inline', element: 'span' },
+          { title: 'My block', element: 'h1' },
+          { title: 'My selector', element: 'div' }
+        ]),
+        sAssertFormatExists('my-inline'),
+        sAssertFormatExists('my-block'),
+        sAssertFormatExists('my-selector')
+      ], onSuccess, onFailure);
+    },
+    {
+      theme: 'silver',
+      toolbar: 'styleselect',
+      base_url: '/project/tinymce/js/tinymce',
+      style_formats: [
+        {
+          name: 'my-inline',
+          title: 'My inline',
+          inline: 'span',
+          classes: [ 'my-inline' ]
+        },
+        {
+          name: 'my-block',
+          title: 'My block',
+          block: 'h1',
+          classes: [ 'my-block' ]
+        },
+        {
+          name: 'my-selector',
+          title: 'My selector',
+          selector: 'p',
+          classes: [ 'my-selector' ]
+        }
+      ]
+    },
+    () => {
```

Comment:
This could be ETA reduced

Reply:
I just copied it from the bespoke test. I can remove that.

Reply:
Hmm, the types do not align. The success to asynctest is `() => void` the one from TinyLoader.setup is `(v?: any, logs?: TestLogs) => void` I guess in TS land it doesn't care but it's a bit odd to pass in arguments to the test function if it some times in the future changes so that it takes an optional argument of any kind this would break.

Reply:
It works in many other places, so it should work here. eg: https://github.com/tinymce/tinymce/blob/25651ccc6ae420615d34e51f20456c444dd6993d/modules/tinymce/src/core/test/ts/browser/FormatterClosestTest.ts#L69

Reply:
I tested this locally and it worked fine for me :shrug: is it possible VS code is attempting to use an older version of TypeScript or something?

Reply:
It works but is it correct:
This is the type of the loaded success function:
https://github.com/tinymce/tinymce/blob/develop/modules/mcagar/src/main/ts/ephox/mcagar/loader/Loader.ts#L6

This is the type of the asynctest success callback:
https://github.com/tinymce/bedrock/blob/master/modules/client/src/main/ts/api/UnitTest.ts#L10

Both v and logs are optional but they will be passed into a function that accepts no arguments. Not sure if TS just treats these as the same since they have no arguments. But I can change this I guess TS will blow up if the nullary function starts taking arguments of a different type.

Reply:
Anyway I eta reduced it anyway. There a also a few places where this has not been done possible for that type issue above.

Reply:
Yeah, you're correct sorry. I misread when looking very quickly :cry: Anyways, it's a bit of a JavaScript oddity I guess, as it'll just ignore anything passed if it doesn't use it. You're also correct that TypeScript will complain if we ever change the `asyncTest` success types, so that's why we consider it safe to reduce this.

Also sorry for not replying to this earlier :disappointed: